geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/process/tube/TubeSelectProcessTest.scala

```diff
@@ -259,8 +262,141 @@ class TubeSelectProcessTest extends Specification {
     }
+
+    "should properly handle geometries crossing the IDL" in {
+      val sftName = "tubeTestType"
+      val sft = SimpleFeatureTypes.createType(sftName, s"type:String,$geotimeAttributes")
+
+      val ds = createStore
+
+      ds.createSchema(sft)
+      val fs = ds.getFeatureSource(sftName)
+
+      val featureCollection = new DefaultFeatureCollection(sftName, sft)
+
+      List("c", "d").foreach { name =>
+        List(1, 2, 3, 4).zip(List(-178, -179, 179, 178)).foreach { case (i, lat) =>
+          val sf = AvroSimpleFeatureFactory.buildAvroFeature(sft, List(), name + i.toString)
+          sf.setDefaultGeometry(WKTUtils.read(f"POINT($lat%d 0)"))
+          sf.setAttribute(DefaultDtgField, new DateTime(f"2011-01-01T00:0$i%d:00Z", DateTimeZone.UTC).toDate)
+          sf.setAttribute("type", name)
+          sf.getUserData()(Hints.USE_PROVIDED_FID) = java.lang.Boolean.TRUE
+          featureCollection.add(sf)
+        }
+      }
+      //Add a point that would be in the buffer if it wrapped around the earth
+      //This point falls between the time of C2 & C3 but west of C2. This point
+      //Would be grabbed if the buffer wrapped the globe due to the IDL.
+      val sf = AvroSimpleFeatureFactory.buildAvroFeature(sft, List(), "d5")
+      sf.setDefaultGeometry(WKTUtils.read(f"POINT(-178.5 0)"))
+      sf.setAttribute(DefaultDtgField, new DateTime(f"2011-01-01T00:02:03Z", DateTimeZone.UTC).toDate)
+      sf.setAttribute("type", "d")
+      sf.getUserData()(Hints.USE_PROVIDED_FID) = java.lang.Boolean.TRUE
+      featureCollection.add(sf)
+
+      // write the feature to the store
+      val res = fs.addFeatures(featureCollection)
+
+      // tube features
+      val tubeFeatures = fs.getFeatures(CQL.toFilter("type = 'c'"))
+
+      // result set to tube on
+      val features = fs.getFeatures(CQL.toFilter("type <> 'c'"))
+
+      // get back type b from tube
+      val ts = new TubeSelectProcess()
+      val results = ts.execute(tubeFeatures, features, null, 1L, 1L, 0.0, 5, "line")
+
+      val f = results.features()
+      while (f.hasNext) {
+        val sf = f.next
+        sf.getAttribute("type") mustEqual "d"
+      }
+
+      results.size mustEqual 4
+    }
+
+    "properly interpolate times and geometries" in {
+      val sftName = "tubeTestType"
+      val sft = SimpleFeatureTypes.createType(sftName, s"type:String,$geotimeAttributes")
+
+      val ds = createStore
+
+      ds.createSchema(sft)
+      val fs = ds.getFeatureSource(sftName)
+
+      val featureCollection = new DefaultFeatureCollection(sftName, sft)
+      //TODO: Make data that actually emulated a track, and two alternate tracks, one offset from the other by a small amount of time
+      val calc = new GeodeticCalculator()
+      //Configure track generation for 1 hour at 10.2 meters per second @ 80 degrees starting at -141.0, 35.0
+      var curPoint = WKTUtils.read("POINT(-141.0 35.0)")
+      val timePerPoint = 150000l  //2.5 minutes in milliseconds
```

Comment:
[style] The `l` might be easier to read as `L` when adjacent to other digits.

Reply:
I'll remember that. Thank you. Fixed.